sos/plugins/filesys.py

```diff
@@ -53,6 +54,10 @@ def setup(self):
         for dev in self.do_regex_find_all(ext_fs_regex, mounts):
                 self.add_cmd_output("dumpe2fs %s %s" % (dumpe2fs_opts, dev))
 
+        if self.get_option('frag'):
+            for dev in self.do_regex_find_all(ext_fs_regex, mounts):
+                self.add_cmd_output("e2freefrag %s" % (dev))
```

Comment:
Why do this loop again here? Just add a conditional on `self.get_option('frag')` inside the existing loop over `ext2` family file systems.